x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java

```diff
@@ -242,7 +242,7 @@ private boolean checkIndexAvailable(ClusterState state) {
             return false;
         }
         final IndexRoutingTable routingTable = state.routingTable().index(metadata.getIndex());
-        if (routingTable == null || routingTable.allPrimaryShardsActive() == false) {
+        if (routingTable == null || routingTable.allSearchShardsActive() == false) {
             logger.debug("Index [{}] is not yet active", aliasName);
             return false;
```

Comment:
This change could have potential issues. The SecurityIndexManager also manages token and profile indices besides the main security index. The profile index uses `fast_refresh` and does not have a separate search shard. I think this change would break most profile APIs, e.g. activation, suggestion. The token index can have separate search shards. But most of its operations are GET and do not rely on search shards. Do we want to block the GET access when the search shard is not available? Similarly, the main security index also sometimes uses the `isAvailable` check to guard GET access which can be blocked by the new check as well.

Reply:
I think it's kind of reasonable to consider the cluster as being in a fairly broken state if these indices do not have a search shard.

However I see that various places are checking `org.elasticsearch.xpack.security.support.SecurityIndexManager.State#indexHealth` to see whether to proceed, which is kind of the same as checking `allPrimaryShardsActive`, so they may need changing too.

Reply:
> The profile index uses fast_refresh and does not have a separate search shard.

I had not considered fast_refresh much since I assumed fast_refresh indices can not have replicas and thus will be always be marked as index and search shards. I asked for some clarification in the comment above and can switch over to use canSearchShard() instead of isSearchable(). 

> Do we want to block the GET access when the search shard is not available? 

Probably not, but it is an edge case. I don't think I know the exact behavior for a GET when a search shard is unavailable ... does it try the real time get, and but could still fail if the translog has already moved on ? how far back w.r.t. time/flush of the translog would the GET actually still work ? 

> to see whether to proceed, which is kind of the same as checking allPrimaryShardsActive, so they may need changing too.

Thanks for calling that one out.  I (or will ask someone) will revisit the security code base looking for other cases where the primary check is insufficient for a future PR.